File: 백준/Gold/1931. 회의실 배정/회의실 배정.java
import java.io.*;
import java.util.*;

public class Main {
    public static void main(String[] args) throws IOException {
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        long N = Long.parseLong(br.readLine());
        long[][] meeting = new long[(int)N][2];

        for (int i = 0; i < N; i++) {
            String[] input = br.readLine().split(" ");
            meeting[i][0] = Long.parseLong(input[0]);
            meeting[i][1] = Long.parseLong(input[1]);
        }

        Arrays.sort(meeting, new Comparator<long[]>() {
            @Override
            public int compare(long[] o1, long[] o2) {
                return o1[1] == o2[1] ? (int)(o1[0] - o2[0]) : (int)(o1[1] - o2[1]);
            }
        });

        int cnt = 1;
        long endTime = meeting[0][1];

        for (int i = 1; i < N; i++) {
            if (endTime <= meeting[i][0]) {
                cnt++;
                endTime = meeting[i][1];
            }
        }

        System.out.println(cnt);
    }
}

File: 백준/Gold/1931. 회의실 배정/회의실 배정.js
const [N, ...input] = require("fs")
  .readFileSync(process.platform === "linux" ? "/dev/stdin" : "input.txt")
  .toString()
  .trim()
  .split("\n");

let meetings = Array.from(Array(Number(N)), () => new Array(2));

for (let i = 0; i < N; i++) {
  [meetings[i][0], meetings[i][1]] = input[i].split(" ");
}

meetings.sort(function (a, b) {
  return a[1] === b[1] ? a[0] - b[0] : a[1] - b[1];
});

let cnt = 0,
  endTime = 0;

meetings.map((meeting) => {
  if (endTime <= Number(meeting[0])) {
    cnt++;
    endTime = Number(meeting[1]);
  }
});

console.log(cnt);
